sunpy/map/maputils.py

```diff
@@ -153,21 +176,66 @@ def contains_full_disk(smap):
     within the field of the view of the instrument (although no emission
     from the disk itself is present in the data.)
     """
-    # Calculate all the edge pixels
-    edges = map_edges(smap)
-    edge_pixels = list(chain.from_iterable([edges[0], edges[1], edges[2], edges[3]]))
-    x = [p[0] for p in edge_pixels] * u.pix
-    y = [p[1] for p in edge_pixels] * u.pix
-
-    # Calculate the edge of the world
-    edge_of_world = smap.pixel_to_world(x, y)
-
-    # Calculate the distance of the edge of the world in solar radii
-    coordinate_angles = np.sqrt(edge_of_world.Tx ** 2 + edge_of_world.Ty ** 2)
-
-    # Test if all the edge pixels are more than one solar radius distant
-    # and that the whole map is not all off disk.
-    return np.all(coordinate_angles > solar_angular_radius(edge_of_world)) and ~is_all_off_disk(smap)
+    # Get the edge pixels
+    top_, bottom, left_hand_side, right_hand_side = map_edges(smap)
+
+    # Coordinates of the edges
+    x, y = _convert_pixels(top_)
+    horizontal1 = smap.pixel_to_world(x, y)
```

Comment:
```suggestion
    horizontal1 = smap.pixel_to_world(_convert_pixels(top_))
```
Is there a better name than `horizontal1`? (The same applies for the other lines.

Reply:
I struggled coming up with these names.  For a N x N image, the `top_ `pixels are (0, N-1),...(N-1, N-1), that is, the second axis value is constant at its maximum value.  A more descriptive name would be `second_axis_value_is_max`.   Following this scheme, we also have `second_axis_value_is_min`, `first_axis_value_is_max` and `first_axis_value_is_min`.  Is this better?